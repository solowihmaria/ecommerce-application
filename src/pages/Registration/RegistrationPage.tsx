import styles from './RegistrationPage.module.scss';
import { RegistrationForm } from '../../components/blocks/RegistrationForm/RegistrationForm';

export const RegistrationPage = () => {
    return (
        <div
            data-testid="test-id-registration-page"
            className={styles.registrationPage}
        >
            <RegistrationForm />
        </div>
    );
};
