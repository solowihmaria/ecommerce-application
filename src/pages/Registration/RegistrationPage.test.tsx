import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { RegistrationPage } from './RegistrationPage';
import { MemoryRouter } from 'react-router-dom';

describe('<RegistrationPage/> component', () => {
    test('Render RegistrationPage component', () => {
        render(
            <MemoryRouter>
                <RegistrationPage />
            </MemoryRouter>
        );
        const registrationPage = screen.getByTestId(
            'test-id-registration-page'
        );
        expect(registrationPage).toBeInTheDocument();
        expect(registrationPage).toMatchSnapshot();
    });
});
