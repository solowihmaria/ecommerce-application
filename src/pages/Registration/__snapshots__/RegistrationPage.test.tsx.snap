// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationPage/> component Render RegistrationPage component 1`] = `
<div
  class="registrationPage"
  data-testid="test-id-registration-page"
>
  <nav
    class="header"
  >
    <div>
      <a
        data-discover="true"
        href="/main"
      >
        <button
          class="button primary"
          type="button"
        >
          Main
        </button>
      </a>
    </div>
    <div
      class="menu"
    >
      <a
        data-discover="true"
        href="/login"
      >
        <button
          class="button primary"
          type="button"
        >
          Sign In
        </button>
      </a>
      <a
        data-discover="true"
        href="/register"
      >
        <button
          class="button primary"
          type="button"
        >
          Sign Up
        </button>
      </a>
    </div>
  </nav>
  <main
    class="registrationContent"
  >
    <div
      class="registrationContainer"
    >
      <form
        class="form"
        data-testid="form-test-id"
      >
        <h2
          class="heading heading--h2 formTitle"
        >
          Registration
        </h2>
        <div
          class="grid2"
        >
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="email"
            >
              Email
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="email"
                  class="input"
                  id="email"
                  name="email"
                  placeholder="Enter your email"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="password"
            >
              Password
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  class="input password"
                  id="password"
                  name="password"
                  placeholder="Create your password"
                  type="password"
                />
                <button
                  aria-label="Show password"
                  class="toggleButton"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 640 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M320 400c-75.85 0-137.25-58.71-142.9-133.11L72.2 185.82c-13.79 17.3-26.48 35.59-36.72 55.59a32.35 32.35 0 0 0 0 29.19C89.71 376.41 197.07 448 320 448c26.91 0 52.87-4 77.89-10.46L346 397.39a144.13 144.13 0 0 1-26 2.61zm313.82 58.1l-110.55-85.44a331.25 331.25 0 0 0 81.25-102.07 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64a308.15 308.15 0 0 0-147.32 37.7L45.46 3.37A16 16 0 0 0 23 6.18L3.37 31.45A16 16 0 0 0 6.18 53.9l588.36 454.73a16 16 0 0 0 22.46-2.81l19.64-25.27a16 16 0 0 0-2.82-22.45zm-183.72-142l-39.3-30.38A94.75 94.75 0 0 0 416 256a94.76 94.76 0 0 0-121.31-92.21A47.65 47.65 0 0 1 304 192a46.64 46.64 0 0 1-1.54 10l-73.61-56.89A142.31 142.31 0 0 1 320 112a143.92 143.92 0 0 1 144 144c0 21.63-5.29 41.79-13.9 60.11z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <fieldset
          class="formGroup fieldset grid3"
        >
          <legend
            class="legend"
          >
            Personal Info
          </legend>
          <div>
            <label
              class="label"
              for="firstName"
            >
              First Name
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="name"
                  class="input"
                  id="firstName"
                  name="firstName"
                  placeholder="Your first name"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div>
            <label
              class="label"
              for="lastName"
            >
              Last Name
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="family-name"
                  class="input"
                  id="lastName"
                  name="lastName"
                  placeholder="Your last name"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div>
            <label
              class="label"
              for="birthDate"
            >
              Date of Birth
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="bday"
                  class="input"
                  id="birthDate"
                  name="birthDate"
                  placeholder="Enter your date of birth"
                  type="date"
                />
              </div>
            </div>
          </div>
        </fieldset>
        <fieldset
          class="fieldset grid4"
        >
          <legend
            class="legend"
          >
            Shipping Address
          </legend>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="shippingStreet"
            >
              Street
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="street-address"
                  class="input"
                  id="shippingStreet"
                  name="shippingStreet"
                  placeholder="Street"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="shippingCity"
            >
              City
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="address-level2"
                  class="input"
                  id="shippingCity"
                  name="shippingCity"
                  placeholder="City"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="shippingCountry"
            >
              Country
              <span
                class="required"
              >
                *
              </span>
            </label>
            <select
              class="select"
              name="shippingCountry"
            >
              <option
                value="france"
              >
                France
              </option>
              <option
                value="germany"
              >
                Germany
              </option>
              <option
                value="italy"
              >
                Italy
              </option>
            </select>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="shippingCode"
            >
              Postal Code
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="postal-code"
                  class="input"
                  id="shippingCode"
                  name="shippingCode"
                  placeholder="Postal code"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="checkboxesContainer"
          >
            <div
              class="checkboxWrapper"
            >
              <div
                class="inputContainer checkbox"
                data-testid="input-test-id"
              >
                <div
                  class="inputWrapper"
                >
                  <input
                    class="input"
                    id="shipping-default"
                    type="checkbox"
                  />
                </div>
              </div>
              <label
                class="label"
                for="shipping-default"
              >
                Set 
                shipping
                 address as default
              </label>
            </div>
            <div
              class="checkboxWrapper"
            >
              <div
                class="inputContainer checkbox"
                data-testid="input-test-id"
              >
                <div
                  class="inputWrapper"
                >
                  <input
                    class="input"
                    id="bill-to-shipping-address"
                    type="checkbox"
                  />
                </div>
              </div>
              <label
                class="label"
                for="bill-to-shipping-address"
              >
                Bill to shipping address
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          class="fieldset grid4"
        >
          <legend
            class="legend"
          >
            Billing Address
          </legend>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="billingStreet"
            >
              Street
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="street-address"
                  class="input"
                  id="billingStreet"
                  name="billingStreet"
                  placeholder="Street"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="billingCity"
            >
              City
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="address-level2"
                  class="input"
                  id="billingCity"
                  name="billingCity"
                  placeholder="City"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="billingCountry"
            >
              Country
              <span
                class="required"
              >
                *
              </span>
            </label>
            <select
              class="select"
              name="billingCountry"
            >
              <option
                value="france"
              >
                France
              </option>
              <option
                value="germany"
              >
                Germany
              </option>
              <option
                value="italy"
              >
                Italy
              </option>
            </select>
          </div>
          <div
            class="formGroup"
          >
            <label
              class="label"
              for="billingCode"
            >
              Postal Code
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class="inputContainer"
              data-testid="input-test-id"
            >
              <div
                class="inputWrapper"
              >
                <input
                  autocomplete="postal-code"
                  class="input"
                  id="billingCode"
                  name="billingCode"
                  placeholder="Postal code"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="checkboxesContainer"
          >
            <div
              class="checkboxWrapper"
            >
              <div
                class="inputContainer checkbox"
                data-testid="input-test-id"
              >
                <div
                  class="inputWrapper"
                >
                  <input
                    class="input"
                    id="billing-default"
                    type="checkbox"
                  />
                </div>
              </div>
              <label
                class="label"
                for="billing-default"
              >
                Set 
                billing
                 address as default
              </label>
            </div>
          </div>
        </fieldset>
        <button
          class="button primary"
          type="submit"
        >
          Create Account
        </button>
        <div
          class="signInRedirect"
        >
          <p
            class="signInText"
          >
            Already have an account?
          </p>
          <a
            class="signInLink"
            data-discover="true"
            href="/login"
          >
            Sign In
          </a>
        </div>
      </form>
    </div>
  </main>
</div>
`;
